### ISSUE TITLE
The remote server returned an error

### ISSUE THREAD

**Issue by bananarama42** [rytsikau/ee.Yrewind#73]
I keep running into this error quite often now. It starts downloading and then at a random percentage it throws this error. After a few tries it usually downloads everything though, but it makes it hard to download anything in a batch as I need to re-run it several times.
![image](https://github.com/user-attachments/assets/c8cf98f9-c0ac-4b2b-828c-65bea50bd9a5)

**Comment by Sion89** [comment]
Yes. Also quite often now.
401 or 403

**Comment by mike365sofresh** [comment]
Yes, same here.  It has been happening randomly for the past several weeks, but today it will error every time.

**Comment by Sion89** [comment]
@rytsikau , we need your help again, nothing is downloading at all, only the first 10-13 MB, then - error "Forbidden 403". It started 20 hours ago.

**Comment by bananarama42** [comment]
I guess there aren't really any updates on this? ☹

**Comment by chengquan100** [comment]
https://github.com/user-attachments/assets/58e36539-a143-4ef5-a4a6-dc9ba933fd4a


Does anyone have this problem? It's been bothering me for a long time :(
`Cannot print exception string because Exception.ToString() failed`

**Comment by Sion89** [comment]
> Untitled.mp4 
> Does anyone have this problem? It's been bothering me for a long time :( `Cannot print exception string because Exception.ToString() failed`

Downloads only the first 10-13MB? I think it's the same 403 error just without a detailed description.

**Please help with alternatives...**

**Comment by Sion89** [comment]
any news?...

**Comment by bananarama42** [comment]
I was hoping it would have magically fixed itself, but still has a 100% fail rate.

**Comment by ststoshev** [comment]
I have discovered the workaround.
When downloading a clip of less than 1min 20 seconds, it manages to finish the streams before it breaks with 403 error. So to download a 10 minute clip, you need to download 10 streams each of 1 minute (for example), then by using ffmpeg to join (-concatenate) the 10 mp4 files in one concatenated big file (without recompression). Under fast internet connection we get around 50 megabytes of data or 1 min: 20 seconds at 1080 pixels resolution. On slower internet, it breaks with 403 and downloads smaller data. So I assume the 403 error is programmed to come up some seconds after the download start.
So we need to automate a large clip to be requested in 1 minute chunks or less, check if all 1-minute intervals are downloaded, and then concatenate the received mp4 files back to the initial clip. Currently I do it manually, but it can easily be automated.

**Comment by chengquan100** [comment]
ANY SOLUTION? @rytsikau 😭😭😭😭
